xtable-utilities/src/main/java/org/apache/xtable/utilities/RunSync.java

```diff
@@ -98,37 +99,116 @@ public class RunSync {
                   + "used for any Iceberg source or target.")
           .addOption(HELP_OPTION, "help", false, "Displays help information to run this utility");
 
-  public static void main(String[] args) throws IOException {
-    CommandLineParser parser = new DefaultParser();
+  static SourceTable sourceTableBuilder(
+      DatasetConfig.Table table,
+      IcebergCatalogConfig icebergCatalogConfig,
+      DatasetConfig datasetConfig,
+      Properties sourceProperties) {
+    Objects.requireNonNull(table, "Table cannot be null");
+    Objects.requireNonNull(datasetConfig, "datasetConfig cannot be null");
```

Comment:
You can use lombok `import lombok.NonNull;` for non null validations to simplify.

Reply:
Actually @rangareddy suggested to use Objects.requireNonNull. So Please let me know if need to change. @vinishjail97

Reply:
We use `lombok.NonNull` in other places in the project, so let's use that ?

Reply:
@vinishjail97  I Made the changes.